packages/core/src/tools/limitModification.ts

```diff
@@ -8,14 +8,15 @@ import { Context, deepClone } from './context'
 export function limitModification<T extends Context>(
   object: T,
   modifiableFieldPaths: string[],
-  modifier: (object: T) => void
-): T {
+  modifier: (object: T) => boolean | void
+): boolean | void {
```

Comment:
is it safe to assume the return type of the modifier?

Reply:
We were assuming that it returns a `void` before, now it can also return a boolean. I don't see the problem with assuming that return type, am I missing something?

Reply:
What about using another parametric type to avoid chosing at this level?

```ts
function limitModification<T extends Context, Result>(
  object: T,
  modifiableFieldPaths: string[],
  modifier: (object: T) => Result
): Result | undefined
```

Reply:
This would be better, I will use this.

Reply:
nice.
however, my point is that since customer can return anything, it can be better to internally handle result value as unknown.

Reply:
Makes sense. I will change it's return type to `unknown`